apps/homescreen/js/app.js

```diff
@@ -540,6 +577,12 @@ const SETTINGS_VERSION = 0;
 
         this.pageHeight = pageHeight;
         this.pendingGridHeight = gridHeight;
+
+        if (!this.visualLoadComplete &&
+            Math.floor(visibleChildren / (this.small ? 4 : 3)) * iconHeight >=
+              scrollHeight) {
+          this.onVisualLoad();
+        }
```

Comment:
That's weird because in 4 columns configuration, the icons on the first page + the ones on the top row of next page appear at the same time.
But in 3 columns mode, I can clearly see the 3 icons of next page loading one after another after the first page icons.
Do you see that too?